pkg/agent/manager/cache/cache.go

```diff
@@ -693,6 +695,15 @@ func (c *Cache) getSelectorIndex(s selector) *selectorIndex {
 	return index
 }
 
+// getSelectorIndexForRead gets the selector index for the selector. If one
+// doesn't exist, nil is returned.
+func (c *Cache) getSelectorIndexForRead(s selector) *selectorIndex {
+	if index, ok := c.selectors[s]; ok {
```

Comment:
We changed the expected behavior from adding a key to the `c.selectors` to directly returning `newSelectorIndex`. Is this in line with expectations, and is there an exception due to no update `c.selectors`?